from PyQt5 import QtWidgets

class Ui_AddLocation(QtWidgets.QDialog):
    def __init__(self, parent=None):
        super().__init__(parent)
        self.setWindowTitle("Add New Location")
        self.resize(900, 600)
        self.main_layout = QtWidgets.QVBoxLayout(self)

        # === General Info ===
        self.general_group = QtWidgets.QGroupBox("General Info")
        form_general = QtWidgets.QFormLayout()
        self.lineEdit_Location = QtWidgets.QLineEdit()
        self.lineEdit_ScanLocation = QtWidgets.QLineEdit()
        self.comboBox_LocationType = QtWidgets.QComboBox()
        self.comboBox_PutawayClass = QtWidgets.QComboBox()
        self.comboBox_PickClass = QtWidgets.QComboBox()
        self.comboBox_RestockClass = QtWidgets.QComboBox()
        form_general.addRow("Location:", self.lineEdit_Location)
        form_general.addRow("Scan Location:", self.lineEdit_ScanLocation)
        form_general.addRow("Location Type:", self.comboBox_LocationType)
        form_general.addRow("Putaway Class:", self.comboBox_PutawayClass)
        form_general.addRow("Pick Class:", self.comboBox_PickClass)
        form_general.addRow("Restock Class:", self.comboBox_RestockClass)
        self.general_group.setLayout(form_general)


        # === Coordinates (Vertical layout) ===
        self.coords_group = QtWidgets.QGroupBox("Physical Coordinates")
        form_coords = QtWidgets.QFormLayout()
        self.lineEdit_Aisle = QtWidgets.QLineEdit()
        self.lineEdit_Bay = QtWidgets.QLineEdit()
        self.lineEdit_Level = QtWidgets.QLineEdit()
        self.lineEdit_Slot = QtWidgets.QLineEdit()
        form_coords.addRow("Aisle:", self.lineEdit_Aisle)
        form_coords.addRow("Bay:", self.lineEdit_Bay)
        form_coords.addRow("Level:", self.lineEdit_Level)
        form_coords.addRow("Slot:", self.lineEdit_Slot)
        self.coords_group.setLayout(form_coords)

        # === Capacity ===
        self.capacity_group = QtWidgets.QGroupBox("Capacity")
        form_capacity = QtWidgets.QFormLayout()
        self.lineEdit_PalletCap = QtWidgets.QLineEdit()
        self.lineEdit_CartonCap = QtWidgets.QLineEdit()
        form_capacity.addRow("Pallet Capacity:", self.lineEdit_PalletCap)
        form_capacity.addRow("Carton Capacity:", self.lineEdit_CartonCap)
        self.capacity_group.setLayout(form_capacity)

        # === Max Dimensions / Weight ===
        self.limits_group = QtWidgets.QGroupBox("Max Dimensions / Weight")
        grid_limits = QtWidgets.QGridLayout()
        self.lineEdit_MaxHeight = QtWidgets.QLineEdit()
        self.comboBox_HeightUOM = QtWidgets.QComboBox()
        self.lineEdit_MaxDepth = QtWidgets.QLineEdit()
        self.comboBox_DepthUOM = QtWidgets.QComboBox()
        self.lineEdit_MaxWidth = QtWidgets.QLineEdit()
        self.comboBox_WidthUOM = QtWidgets.QComboBox()
        self.lineEdit_MaxWeight = QtWidgets.QLineEdit()
        self.comboBox_WeightUOM = QtWidgets.QComboBox()
        grid_limits.addWidget(QtWidgets.QLabel("Max Height:"), 0, 0)
        grid_limits.addWidget(self.lineEdit_MaxHeight, 0, 1)
        grid_limits.addWidget(self.comboBox_HeightUOM, 0, 2)
        grid_limits.addWidget(QtWidgets.QLabel("Max Depth:"), 1, 0)
        grid_limits.addWidget(self.lineEdit_MaxDepth, 1, 1)
        grid_limits.addWidget(self.comboBox_DepthUOM, 1, 2)
        grid_limits.addWidget(QtWidgets.QLabel("Max Width:"), 2, 0)
        grid_limits.addWidget(self.lineEdit_MaxWidth, 2, 1)
        grid_limits.addWidget(self.comboBox_WidthUOM, 2, 2)
        grid_limits.addWidget(QtWidgets.QLabel("Max Weight:"), 3, 0)
        grid_limits.addWidget(self.lineEdit_MaxWeight, 3, 1)
        grid_limits.addWidget(self.comboBox_WeightUOM, 3, 2)
        self.limits_group.setLayout(grid_limits)

        # === Proximity ===
        self.prox_group = QtWidgets.QGroupBox("Proximity Settings")
        form_prox = QtWidgets.QFormLayout()
        self.comboBox_BlockCode = QtWidgets.QComboBox()
        self.comboBox_ProxIN = QtWidgets.QComboBox()
        self.comboBox_ProxOUT = QtWidgets.QComboBox()
        self.lineEdit_PnD1 = QtWidgets.QLineEdit()
        self.lineEdit_PnD2 = QtWidgets.QLineEdit()
        form_prox.addRow("Block Code:", self.comboBox_BlockCode)
        form_prox.addRow("Proximity In:", self.comboBox_ProxIN)
        form_prox.addRow("Proximity Out:", self.comboBox_ProxOUT)
        form_prox.addRow("PnD Location 1:", self.lineEdit_PnD1)
        form_prox.addRow("PnD Location 2:", self.lineEdit_PnD2)
        self.prox_group.setLayout(form_prox)

        # === Flags ===
        self.flags_group = QtWidgets.QGroupBox("Status Flags")
        layout_flags = QtWidgets.QVBoxLayout()
        self.checkBox_BlackHole = QtWidgets.QCheckBox("Black Hole")
        self.checkBox_Assigned = QtWidgets.QCheckBox("Has assign flag")
        self.checkBox_Content = QtWidgets.QCheckBox("Has content flag")
        self.checkBox_Pending = QtWidgets.QCheckBox("Has pending flag")
        layout_flags.addWidget(self.checkBox_BlackHole)
        layout_flags.addWidget(self.checkBox_Assigned)
        layout_flags.addWidget(self.checkBox_Content)
        layout_flags.addWidget(self.checkBox_Pending)
        self.flags_group.setLayout(layout_flags)

        # === Footer ===
        footer_layout = QtWidgets.QHBoxLayout()
        self.lineEdit_LastTouched = QtWidgets.QLineEdit()
        self.lineEdit_LastTouched.setReadOnly(True)
        self.lineEdit_PalletsQty = QtWidgets.QLineEdit()
        self.lineEdit_PalletsQty.setReadOnly(True)
        self.lineEdit_CartonsQty = QtWidgets.QLineEdit()
        self.lineEdit_CartonsQty.setReadOnly(True)
        footer_layout.addWidget(QtWidgets.QLabel("Last touched:"))
        footer_layout.addWidget(self.lineEdit_LastTouched)
        footer_layout.addSpacing(30)
        footer_layout.addWidget(QtWidgets.QLabel("Active Pallets QTY:"))
        footer_layout.addWidget(self.lineEdit_PalletsQty)
        footer_layout.addWidget(QtWidgets.QLabel("Active Cartons QTY:"))
        footer_layout.addWidget(self.lineEdit_CartonsQty)

        # === Distribuci√≥n en columnas ===
        left_col = QtWidgets.QVBoxLayout()
        left_col.addWidget(self.general_group)
        left_col.addWidget(self.capacity_group)
        left_col.addWidget(self.limits_group)

        right_col = QtWidgets.QVBoxLayout()
        right_col.addWidget(self.coords_group)
        right_col.addWidget(self.prox_group)
        right_col.addWidget(self.flags_group)

        main_split = QtWidgets.QHBoxLayout()
        main_split.addLayout(left_col, 3)
        main_split.addLayout(right_col, 2)

        self.main_layout.addLayout(main_split)

        # Footer ocupa todo el ancho abajo
        self.main_layout.addSpacing(10)
        self.main_layout.addLayout(footer_layout)